python/python/bystro/ancestry/ancestry_types.py

```diff
@@ -0,0 +1,158 @@
+"""Classes for common shapes of data in ancestry."""
+
+import attr
```

Comment:
I think we want to import `attrs` not `attr`. The authors recommend it, though I don't get a good sense of why besides performance and sensible defaults.

Reference: https://www.attrs.org/en/stable/names.html

Instead of `attr.s` we would use `attrs.frozen`